src/iomusxml.cpp

```diff
@@ -641,9 +641,11 @@ bool MusicXmlInput::ReadMusicXml(pugi::xml_node root)
 
     if (!m_ClefChangeStack.empty()) {
         for (musicxml::ClefChange iter : m_ClefChangeStack) {
-            LogWarning("MusicXML import: Clef change at measure %s, staff %d, time %d not inserted.",
-                iter.m_measureNum.c_str(), iter.m_staff->GetN(), iter.m_scoreOnset);
+            if (iter.isFirst)
```

Comment:
Sorry, it is difficult for me to have an overview, but don't you have to delete the clef? That is, something like:
```
delete iter.m_clef;
```

Reply:
The ClefChange object just has pointers inside. And, in all my testing files, there was no single case of an uninserted clef change, so I cannot test that situation.

I checked with Xcode analyze and found just another potential leak in existing code: 
<img width="267" alt="grafik" src="https://user-images.githubusercontent.com/45632600/57913025-fe13ab80-788b-11e9-8977-2e6718acf80d.png">

Reply:
The memory leak in iomusicxml.cpp is here: https://github.com/rism-ch/verovio/blob/c83ac38e0fc0414fda9437915ef18f086b7caa1e/src/iomusxml.cpp#L1148 (or line 1198 in my version)

Reply:
Correct. And it has been there for a while...
https://github.com/rism-ch/verovio/commit/a6662949d0881bf763d8bf2a221d712f4d1c278c

(I will fix the one in view_slur.cpp, thanks for running the analyser)

> The ClefChange object just has pointers inside. And, in all my testing files, there was no single case of an uninserted clef change, so I cannot test that situation.

Yes, this is the question here: was the problem added to the document tree (e.g., through `Layer::AddChild`) or not? If not, then the object needs to be deleted explicitly.

Verovio is pretty robust because the memory is managed through its tree structure. That is, once an object is added as a child within the (Doc) tree, the parent will be in charge of deleting it. But we still need to make sure all objects created on the heap with `new` are all added to the tree. These memory leaks will be difficult to spot. The analyzer seem to be a good job but will probably not find all of them. I sometimes run the Apple Instruments that would probably spot this one if you have the corresponding data to test it.

Reply:
I am now looking again and it is looking good. I think I misunderstood what was happening because of the additional loop where I thought you where trying to delete unused clefs.

Reply:
Many thanks for merging! The wonderful code structure makes it easy to contribute.

Reply:
> Yes, this is the question here: was the clef added to the document tree (e.g., through Layer::AddChild) or not? If not, then the object needs to be deleted explicitly.

I will look into that to understand where it is added.

Reply:
So, as it seems, the layer there gets added by the stacked `SelectLayer()` method. This should be fine then.